test/general-tests/constructor-test/check.ts

```diff
@@ -1,8 +1,38 @@
-import { hardhatStarknetCompileDeprecated, hardhatStarknetTest } from "../../utils/cli-functions";
+import path from "path";
+import { hardhatStarknetCompile, hardhatStarknetTest } from "../../utils/cli-functions";
+import { copyFileSync } from "fs";
+import { assertContains, rmrfSync } from "../../utils/utils";
 
-hardhatStarknetCompileDeprecated(
-    "contracts/contract.cairo contracts/simple_storage.cairo contracts/empty_constructor.cairo".split(
-        " "
-    )
-);
+const prefix = path.join(__dirname);
+const sourcesPath = "cairo1-contracts";
+
+const duplicateConstructorContract = "duplicate_constructor.cairo";
+const contract1Path = path.join(sourcesPath, duplicateConstructorContract);
```

Comment:
I think the descriptive naming should have been applied to the path variables as well. But I think we don't need the path variables really. We can use a loop instead of duplicating all the path joinings and copyFileSyncs. Maybe even the contract name variables aren't needed (except for this one because it has to be deleted later).

Reply:
https://github.com/0xSpaceShard/starknet-hardhat-plugin/pull/377/commits/81d8dd0f8431b12eac70fcdd226edc426bae3432